File: config/src/main/java/com/sybd/znld/config/ProjectConfig.java
package com.sybd.znld.config;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import java.time.Duration;

//@Component
@ConfigurationProperties(prefix = "project")
@Getter @Setter
public class ProjectConfig {
    public String name;
    public String cachePrefix;
    public Duration cacheOfNullExpirationTime;
    public Duration cacheOfCaptchaExpirationTime;
    public Duration auth2TokenExpirationTime;
}

File: znld-web/src/main/java/com/sybd/znld/config/ProjectConfig.kt
package com.sybd.znld.config;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

import java.time.Duration;

@Component
@ConfigurationProperties(prefix = "project")
class ProjectConfig {
    lateinit var name: String;
    lateinit var cachePrefix: String ;
    lateinit var cacheOfNullExpirationTime: Duration;
    lateinit var cacheOfCaptchaExpirationTime: Duration;
    lateinit var auth2TokenExpirationTime: Duration;
}
